main.go

```diff
@@ -27,19 +27,36 @@ Options:
   -h --help                     Show this screen.
   -v --version                  Show version.`
 
+	return docopt.Parse(usage, nil, true, "ury-listd-go 0.0", false)
+}
+
+func main() {
 	logger := log.New(os.Stdout, "", log.Lshortfile)
-	args, err := docopt.Parse(usage, nil, true, "ury-listd-go 0.0", false)
+	args, err := parseArgs()
 	if err != nil {
 		logger.Fatal("Error parsing args: " + err.Error())
 	}
+
+	// Set up server listener
+	serverCh := make(chan string)
+	server, err := MakeServer(args["--addr"].(string), args["--port"].(string), serverCh)
+	if err != nil || server == nil {
```

Comment:
Surely if err is not nil server won't be? And even if err is nil but server is not you'll print nil?